minimalkv/net/gcstore.py

```diff
@@ -55,197 +20,36 @@ def __init__(
         bucket_creation_location: str = "EUROPE-WEST3",
         project=None,
     ):
+        if isinstance(credentials, str):
+            # Parse JSON from path to extract project name
+            try:
+                with open(credentials) as f:
+                    credentials_dict = json.load(f)
+                    project = project or credentials_dict["project_id"]
+            except (FileNotFoundError, json.JSONDecodeError) as error:
+                print(
+                    f"Could not get the project name from the credentials file: {error}"
+                )
```

Comment:
is it okay for the `project` attribute to be `None`? Or will this mess with code execution further down the road?

Reply:
Yes, having the project name is important when running against live GCS, as it is required to create new buckets.
I added a warning message, because the bucket might already exist, which wouldn't be a problem.